server/controllers/productController.js

```diff
@@ -15,48 +25,90 @@ class ProductController {
         } catch (e) {
             next(ApiError.badRequest((e.message)))
         }
-
     }
-    async getAll(req, res) {
-        let { categoryId, limit = 10 , page = 1 } = req.body
-        let offset = page * limit - limit
-        let products
-
-        if (categoryId) {
-            products = await Product.findAndCountAll({ where: {categoryId}, limit, offset })
-        } else {
-            products = await Product.findAndCountAll({ limit, offset })
-        }
 
-        return res.json(products)
+    async getAll(req, res, next) {
+        try {
+            let { categoryId, limit = 10 , page = 1 } = req.body
+            let offset = page * limit - limit
+            let products
+
+            const query = categoryId ? {where: {categoryId}, limit, offset } : { limit, offset };
+
+            products = await Product.findAndCountAll(query)
 
+            return res.json(products)
+        } catch (e) {
+            next(ApiError.badRequest((e.message)))
+        }
     }
 
-    async getOne(req, res) {
-        const {id} = req.params
+    //todo come up with a good way to display the rating
+    async getOne(req, res, next) {
+        try {
+            const { id } = req.params
+            const { user } = req.body
+
+            const product = await Product.findOne(
+                {
+                    where: { id },
+                    include: [{ model: ProductInfo, as: 'info' }, { model: Rating, as: 'rateInfo' }]
+                }
+            )
+
+            const calculateRating = (ratings) => {
+                let rating = 0
+                if (ratings) {
+                    ratings.forEach(i => {
+                        rating += i
+                    })
+                    rating = rating / ratings.length
+                }
+
+                return rating
+            }
 
-        const product = await Product.findOne(
-            {
-                where: {id},
-                include: [{model: ProductInfo, as: 'info'}]
+            const parsedProduct = JSON.parse(JSON.stringify(product))
+            const newProduct = {...parsedProduct}
+
+            if (!parsedProduct) {
+                return next(ApiError.notFound('Нет такой трубы, дружок!)'))
+            }
+
+            if (parsedProduct.rateInfo.length){
+                newProduct.rateInfo = {
+                    isRated: parsedProduct.rateInfo.reduce((total, item) => {
+                        if (item.user === Number(user)){
+                            total = true
+                        }
+                        return total
+                    }, false),
+                    rate: calculateRating(product.rateInfo.map(item=> item.rate))
+                }
+            } else {
+                newProduct.rateInfo = {
+                    isRated: false,
+                    rate: 0
+                }
             }
-        )
 
-        return res.json(product)
+            return res.json(newProduct)
```

Comment:
It would be better to write it like this:
return res.json({...product, isRated: product.rateInfo.length ? ... : false, rate: product.rateInfo.length ? ... : 0})

Or even create a new function and:
const getRateData = (product) => {
  if (product.rateInfo.length) {
    return {
      isRated: ...,
      rate: ...
    }
  } else { 
     return {
       isRated: false,
       rate: 0
   }
  }
}